<?xml version="1.0"?>

<launch>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find igvc_gazebo)/config/igvc_joint_control.yaml" command="load"/>

  <!-- load controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner"
    args="$(find igvc_gazebo)/config/igvc_joint_control.yaml"/>

  <node name="igvc_control" pkg="igvc_gazebo" type="control">
      <!-- PID values for wheel speed -->
      <param name="speed_P_left" value="0.5"/>
      <param name="speed_P_right" value="0.5"/>
      <param name="speed_D_left" value="3.0"/>
      <param name="speed_D_right" value="3.0"/>
      <param name="wheel_radius" value="0.3429"/>
      <param name="rate" value="60.0"/>
  </node>

</launch>
